sdk/compiler/daml-extension/src/extension.ts

```diff
@@ -187,6 +119,134 @@ export async function activate(context: vscode.ExtensionContext) {
+interface IdeManifest {
+  [multiPackagePath: string]: { [envVarName: string]: string };
+}
+
+function parseIdeManifest(path: string): IdeManifest | null {
+  try {
+    const json = JSON.parse(fs.readFileSync(path, "utf8"));
+    const manifest: IdeManifest = {};
+    for (const entry of json) {
+      manifest[entry["multi-package-directory"]] = entry.environment;
+    }
+    return manifest;
+  } catch (e) {
+    console.log(e);
+    return null;
+  }
+}
+
+async function fileExists(path: string) {
+  return new Promise(r => fs.access(path, fs.constants.F_OK, e => r(!e)));
+}
+
+async function startLanguageServers(context: ExtensionContext) {
+  const config = vscode.workspace.getConfiguration("daml");
+  const consent = await getTelemetryConsent(config, context);
+  const rootPath = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
+  if (!rootPath) throw "Couldn't find workspace root";
+  const multiIDESupport = config.get("multiPackageIdeSupport");
+
+  var isGradleProject = false;
+  const gradlewExists = await fileExists(rootPath + "/gradlew");
+  // Of the 3 direnv extensions I found, mkhl.direnv is most popular and only one that populates environment variables for other extensions
+  // As such, it is the one we recommend.
+  const envrcExistsWithoutExt =
+    (await fileExists(rootPath + "/.envrc")) &&
+    vscode.extensions.getExtension("mkhl.direnv") == null;
+
+  if (envrcExistsWithoutExt) {
+    const warningMessage =
+      "Found an .envrc file but the recommended direnv VSCode extension was not installed. Daml IDE may fail to start due to missing environment." +
+      "\nWould you like to install this extension or attempt to continue without it?";
+    const installAnswer = "Install recommended direnv extension";
+    const doNotInstallAnswer = "Attempt to continue without";
+    const doNotInstallkey = "no-install-direnv";
+
+    // Don't ask if we previously selected doNotInstallAnswer
+    if (!context.workspaceState.get(doNotInstallkey, false)) {
+      const selection = await vscode.window.showWarningMessage(
+        warningMessage,
+        installAnswer,
+        doNotInstallAnswer,
+      );
+
+      if (selection == doNotInstallAnswer)
+        context.workspaceState.update(doNotInstallkey, true);
+      else if (selection == installAnswer) {
+        await vscode.commands.executeCommand("extension.open", "mkhl.direnv");
+        vscode.window.showInformationMessage(
+          "After installing, reload the VSCode window or restart the extension host through the command palette.",
+        );
+        return;
+      }
+    }
+  }
+
+  if (multiIDESupport && gradlewExists) {
+    const generateIdeManifestCommand =
+      rootPath + '/gradlew run --args="--project-dir ' + rootPath + '"';
+    try {
+      await util.promisify(child_process.exec)(generateIdeManifestCommand, {
+        cwd: rootPath,
+      });
+      isGradleProject = true;
+    } catch (e) {
+      // Probably do something here
+      console.log(e);
```

Comment:
If this fails, we should show a proper error, but consider that it could simply mean we're in a gradle project without the integration, perhaps warn and try to continue?